core/arch/arm/kernel/thread.c

```diff
@@ -1199,7 +1199,8 @@ static uint32_t rpc_cmd_nolock(uint32_t cmd, size_t num_params,
 
 	assert(arg && carg && num_params <= THREAD_RPC_MAX_NUM_PARAMS);
 
-	plat_prng_add_jitter_entropy_norpc();
+	if (cmd != OPTEE_MSG_RPC_CMD_WAIT_QUEUE)
```

Comment:
Ha! That's a tricky one. I think this deserves a comment like "Break recursion in case plat_prng_add_jitter_entropy_norpc() invokes a mutex" or something like that.

Also I would recommend commenting the prototype for plat_prng_add_jitter_entropy_norpc():
```diff
diff --git a/core/include/tee/tee_cryp_utl.h b/core/include/tee/tee_cryp_utl.h
index 28b501d..76e852a 100644
--- a/core/include/tee/tee_cryp_utl.h
+++ b/core/include/tee/tee_cryp_utl.h
@@ -50,6 +50,10 @@ TEE_Result tee_aes_cbc_cts_update(void *cbc_ctx, void *ecb_ctx,
 
 TEE_Result tee_prng_add_entropy(const uint8_t *in, size_t len);
 void plat_prng_add_jitter_entropy(void);
+/*
+ * The _norpc version must not invoke Normal World, or infinite recursion
+ * may occur. As an exception however, using mutexes is allowed.
+ */
 void plat_prng_add_jitter_entropy_norpc(void);
 
 #endif
```